### PR TITLE
Add filter for Kotlin coroutines

### PR DESCRIPTION
**Pull request by Godin**

### CODE REVIEW

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+				output.ignore(ignore.get(i), ignore.get(i + 1));
+			}
+		}
+
+		/**
+		 * Moves {@link #cursor} to next instruction if it is
+		 * {@link TypeInsnNode} with given opcode and operand, otherwise sets it
+		 * to <code>null</code>.
+		 */
+		private void nextIsType(final int opcode, final String desc) {
```
Comment (marchof) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java]: @Godin Pull this up to `AbstractMatcher` and add a unit test?

Comment (Godin) [reply org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java]: Done, moreover - replaced `nextIsNew` by `nextIsType` :wink:

File: org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+import org.objectweb.asm.tree.TableSwitchInsnNode;
+import org.objectweb.asm.tree.TypeInsnNode;
```
Comment (marchof) [inline org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java]: Unused import gives a compiler warning.

Comment (Godin) [reply org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinCoroutineFilter.java]: Removed.

**Pull request merged**